### PR TITLE
Implement reservation

### PR DESCRIPTION
**Pull request by MStodolkiewicz**
Add Reservation model and controller

Add resources and references

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+class ReservationsController < ApplicationController
+  include Pundit::Authorization
+
+  before_action :set_reservation, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # GET /reservations or /reservations.json
```
Comment (h4ppyr0gu3) [inline app/controllers/reservations_controller.rb]: please remove unnecessary comments
code should be self documenting
ie the function name describes itself

File: app/models/reservation.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Reservation < ApplicationRecord
+  belongs_to :seance
+  belongs_to :user
+
+  validates :status, presence: true
+  enum :status, %i[created confirmed canceled]
```
Comment (h4ppyr0gu3) [inline app/models/reservation.rb]: use explicit enums where possible
it makes it more readable for other developers
```suggestion
  enum status: { created: 0, confirmed: 1, canceled: 2 }
```

File: app/models/user.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 class User < ApplicationRecord
+  has_many :reservations, dependent: :destroy
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
   enum role: %i[user manager admin]
```
Comment (h4ppyr0gu3) [inline app/models/user.rb]: consider an explicit enum
enum role: { user: 0, manager: 1, admin: 2 }
we then know what the database fields are more explicitly

File: app/controllers/reservations_controller.rb (hunk 1 above)
Comment (MStodolkiewicz) [reply app/controllers/reservations_controller.rb]: Removed them from controllers :)

File: app/models/reservation.rb (hunk 2 above)
Comment (MStodolkiewicz) [reply app/models/reservation.rb]: Changed both enums, I'm gonna use explicit enums from now on :D.

**Pull request merged**